Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +149,181 @@ public static class CustomEntitySearch
+                    }
+                    else
+                    {
+                        escapedWord.Append(wordCharArray[currWordCharIndex]);
+                    }
+                }
+                if (!wordCharArray.Last().IsDelineating() && !substringMatch)
+                    escapedWord.Append(@"\b");
+                escapedWord.Append("))");
+                string pattern = (caseSensitive) ? @"(?x)" + escapedWord : @"(?ix)" + escapedWord;
+                
+                MatchCollection entityMatch = Regex.Matches(text, pattern, RegexOptions.Compiled, TimeSpan.FromSeconds(MaxRegexEvalTime));
+                if (entityMatch.Count != 0)
+                {
+                    foreach (Match match in entityMatch)
+                    {
+                        entities.Add(
+                            new Entity
+                            {
+                                Category = "customEntity",
+                                Value = match.Groups[1].Value,
+                                Offset = match.Index,
+                                Confidence = 0
+                            });
+                    }
+                    entitiesFound.Add(checkMatch);
+                }
+            }
+            else
+            {
+                List<int> startPointersInText = new List<int> { 0 };
+                List<int> endPointersInText = new List<int>();
+                string textCharArray = (caseSensitive) ? text : text.ToLower(CultureInfo.CurrentCulture);
+                for (int currTextCharIndex = 0; currTextCharIndex < textCharArray.Length; currTextCharIndex++)
+                {
+                    if (textCharArray[currTextCharIndex].IsDelineating())
+                    {
+                        if (currTextCharIndex + 1 < textCharArray.Length && !textCharArray[currTextCharIndex + 1].IsDelineating())
+                            startPointersInText.Add(currTextCharIndex + 1);
+                        if (currTextCharIndex - 1 >= 0 && !textCharArray[currTextCharIndex - 1].IsDelineating())
+                            endPointersInText.Add(currTextCharIndex - 1);
+                    }
+                }
+                endPointersInText.Add(textCharArray.Length - 1);
+
+                double[] minLevenshteinDistance = new double[startPointersInText.Count];
+                int[] endofMatchInTextPointer = new int[startPointersInText.Count];
+                for (int startIndex = 0; startIndex < startPointersInText.Count; startIndex++)
+                {
+                    minLevenshteinDistance[startIndex] = -1;
+                    for (int endIndex = startIndex; endIndex < endPointersInText.Count; endIndex++)
+                    {
+                        double distance = DamerauLevenshteinCalculation(textCharArray.Substring(startPointersInText[startIndex],
+                            endPointersInText[endIndex] - startPointersInText[startIndex] + 1), wordCharArray);
```

Comment:
>endPointersInText[endIndex] - startPointersInText[startIndex] + 1 [](start = 28, length = 65)

you can short circuit out before doing the calculation if the length of this string is too big